src/server.ts

```diff
@@ -10,6 +13,21 @@ function getRovers(res: any) {
       .then((e) => res.send(e.data));
   }
 
+  function getPhotos(rovername: rover, cameratype: camera, res: any) {
+    let data = axios
+      .get(
+        `https://api.nasa.gov/mars-photos/api/v1/rovers/${rovername}/photos?sol=1000&camera=${cameratype}&api_key=6R1WgjArW8R3yvTqvtv1iljThtiB9ANQVIn01PJm`
+      )
+      .then((e) => {
+        let photos = e.data.photos; // list of photo objects (list of strings technically)

```

Comment:
Code should largely explain itself (you can read more about this in the Clean Code module). We want to reserve comments for when code is behaving weirdly and it needs to be explained, if you find that your code requires a comment then its usually an indication to reconsider the approach in case there's a way to write it clearer.

In this case, you can add a type definition to photos (e.g. `photos: Photo[] = e.data.photos`

Reply:
Oooh interesting, yeah I reckon its because the API response is a string list and there _should_ be a way for you to convert it into an object of your desired type. Try using `json.parse()` on each string in the array and seeing if that conforms to your `Photo` type. If this doesn't work, see what the error says as you may be missing or having incorrectly named stuff in your Type definition.

Something that gets used on a lot of projects for JSON validation is a library such as [zod](https://zod.dev/) which lets you define JSON schema and then do `schema.parse(<response>)` to validate and confirm your object (or raise an error if it fails validation). You could try this out if the basic parsing doesn't work :D